tests/Functional/Command/ChangePasswordCommandTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Tests\Functional\Command;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sonata\UserBundle\Model\UserInterface;
+use Sonata\UserBundle\Tests\App\AppKernel;
+use Sonata\UserBundle\Tests\App\Entity\User;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Console\Tester\CommandTester;
+
+class ChangePasswordCommandTest extends KernelTestCase
+{
+    private CommandTester $commandTester;
+
+    protected function setUp(): void
+    {
+        static::bootKernel();
+
+        $this->commandTester = new CommandTester(
+            (new Application(static::createKernel()))->find('sonata:user:change-password')
+        );
+    }
+
+    public function testThrowsWhenUserDoesNotExist(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+
+        $this->commandTester->execute([
+            'username' => 'sonata-user-test',
+            'password' => 'password',
+        ]);
+    }
+
+    public function testChangesUserPassword(): void
+    {
+        $user = $this->prepareData('sonata-user-test', 'old_password');
+
+        static::assertSame($user->getPassword(), 'old_password');
+
+        $this->commandTester->execute([
+            'username' => 'sonata-user-test',
+            'password' => 'new_password',
+        ]);
+
+        $user = $this->refreshUser($user);
+
+        // static::assertSame($user->getPassword(), 'new_password');
```

Comment:
The password is not really changed.

That's because the plainPassword field is not managed by Doctrine, so if we only modify this it is unable to know something changed and trigger the preUpdate lifecycle event.

On FOSUserBundle , the UserManager "save" method does not only save the object but also updated the canonical fields and the password. Should we do the same?

It will also affect the UserAdmin when you only change password, so we should do the same there.

wdyt @VincentLanglet ? At the end we have some duplicated things. The preUpdate event and the code on the manager and the admin.

Reply:
Fixed by adding the updatePassword to the ModelManager, not sure if I like it a lot but it works.